src/ActivityPub/ActorHandle.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\ActivityPub;
+
+class ActorHandle
+{
+    public const HANDLE_PATTERN = '/^(?P<prefix>[@!])?(?P<user>[\w\-\.]+)@(?P<host>[\w\.\-]+)(?P<port>:[\d]+)?$/';
+
+    public function __construct(
+        public ?string $prefix = null,
+        public ?string $username = null,
+        public ?string $host = null,
+        public ?int $port = null,
+    ) {
+    }
+
+    public function __toString(): string
+    {
+        return $this->formatWithPrefix($this->prefix);
+    }
+
+    public static function parse(string $handle): ?static
+    {
+        if (preg_match(static::HANDLE_PATTERN, $handle, $match)) {
+            $new = new static(
+                $match['prefix'] ?? null,
+                $match['user'],
```

Comment:
It's not really a `user` right? It's a group in AP or community. Maybe change the regex `<user>`

Reply:
Ow wait you gonna use this for user? No.. you shoudn't. `!` syntax is used for finding the community or group.

Reply:
it's the username part of the handle, whether it identify a user, group or other kinds of actor that can be expressed in the format of `username@domain[:port]`

perhaps I could at least rename this to just `name` to avoid confusion and avoid implying extra connection between the handle and what kind of actor it probably points to